### PR TITLE
fix(#8242): fix tooltip

### PR DESCRIPTION
**Pull request by latin-panda**
# Description

This work focuses on the date tooltip only.
- Removes broken code that was using the tooltip from the Tour Guide.
- Adds support for the tooltip in mobile: Use the title attribute value and put it in a pseudo element of CSS.
- Desktop will use the native browser tooltip now. 

medic/cht-core#8242

Recap of tested scenarios: [video](https://github.com/medic/cht-core/assets/66472237/05a143da-62bb-4f38-b591-91cf81ea3405)
- Chrome desktop
- Android 10 and 12

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8242-fix-date-tooltip/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8242-fix-date-tooltip/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8242-fix-date-tooltip/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by latin-panda** [comment]
@ngaruko the unit tests are already asserting the correctness of the `title` attribute in the HTML. But I didn't find a way to add an e2e test to check that the tooltip actually **displays** and then assert the text displayed. Because native browser tooltips (solution for desktop) and pseudo-elements from CSS (tooltip solution for mobile) are inaccessible, as far I can see.
What do you suggest for this?

**Comment by latin-panda** [comment]
Hi @m5r can you please review this PR?

**Comment by latin-panda** [comment]
Thanks, @ngaruko. To ensure we are on the same page, we're looking to check the tooltip's actual display, for example, when pressing in mobile, to catch this issue from happening again. We're already checking for the HTML title property in some other places, and it had the correct value, but it didn't catch this bug before :(

**Comment by latin-panda** [comment]
@ngaruko Thanks for the sample.
That doesn't work because `data-original-title` is something from the [bootstrap tour guide script](https://github.com/medic/cht-core/blob/46bb03fcef3bde882a376bb488608a1e577e7ffd/webapp/src/js/bootstrap-tour-standalone.js#L429) that [we removed](https://github.com/medic/cht-core/commit/d2593c4f80a98d4a816afd87278d247d2ee9d0f4#diff-ea9b72b1bbe09c7e93be421587d0dcbd0bf46c285395e662d1a1989cbf56ee39) some weeks ago. That's why we have this bug 😅

We're not going to add it back. Instead, we're using the native HTML title property for the desktop and resized screens, I'm pretty sure this one is working fine, and we have test coverage for the code that generates the date's HTML.
Screenshot 2023-06-06 at 7 49 29 pm

I was more interested in covering the solution for mobile in case this bug happens again; we are [using CSS](https://github.com/medic/cht-core/pull/8290/files#diff-60ebae2b764c2eb81efbfa0d98011c16f8e6d273e232567935c215226cdbe6c1R15) to take the text from the title attribute and display it in a pseudo-element, it's light in processing and less invasive:  
Screenshot 2023-06-06 at 7 33 46 pm
But in the end, I didn't find a way. The best I could do was manually test it, and I have attached the scenarios in a video on this PR's description. 
Is it enough for this work?

**Comment by latin-panda** [comment]
@m5r I got the e2e working now! Thanks for helping me out :)

**Review by m5r — APPROVED** [review]
Nice work Jenni, I left one question about the regex but besides that it's good to go :)

File: webapp/src/css/tooltip.less [hunk 1]
```diff
@@ -0,0 +1,43 @@
+@import 'variables';
+
+@media (pointer: coarse), (hover: none) {
+  /**
+   * Makes native “title” accessible from mobile devices if the element has tabindex=“0"
```
[the remark this answers is not in the document]
Comment (latin-panda) [reply webapp/src/css/tooltip.less:5]: There are smart people in Stackoverflow 😄

File: webapp/src/css/tooltip.less [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    [title]:focus:first-child::before {
+      top: 28px;
+    }
+  }
+
+  .item-summary .relative-date:focus {
+    min-width: 140px;
+  }
+
+  .sync-status .last-sync:focus-within {
```
[the remark this answers is not in the document]
Comment (latin-panda) [reply webapp/src/css/tooltip.less:40]: Thank you!

**Pull request merged**